pkg/iniparser/parser.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+var (
+	ErrSectionNotExists = errors.New("Section doesn't exist")
+	ErrKeyNotExists     = errors.New("cKey doesn't exist")
+)
+
+type Iniparser struct {
+	sections map[string]map[string]string
+}
+
+// a method to load the data from the a string that holds the ini structure into the iniparser
+func (i *Iniparser) LoadFromString(config string) {
+	i.sections = map[string]map[string]string{}
+	lines := strings.Split(config, "\n")
+	section := ""
+	m := map[string]string{}
+
+	for _, line := range lines {
+		line := strings.TrimSpace(line)
+		if len(line) != 0 {
+			if line[0] == '[' {
+				section = strings.Replace(line, "[", "", 1)
+				section = strings.Replace(section, "]", "", 1)
+				m = make(map[string]string)
+			} else if line[0] != '#' && line[0] != ';' {
+				pair := strings.Split(line, " = ")
```

Comment:
- you should split over `=` (no spaces before or after)
- `Split` will split all instances of the string so `key=val1=val2` would be key,val1,val2 and the final key and value would be `key` and `val1` which is not the case it should be `key` and `val1=val2`